packages/dashboard-backend/src/devworkspaceClient/services/dockerConfigApi.ts

```diff
@@ -28,59 +24,58 @@ const SECRET_LABELS = {
   'controller.devfile.io/watch-secret': 'true',
 };
 const DOCKER_CONFIG_API_ERROR_LABEL = 'CORE_V1_API_ERROR';
+const EMPTY_DOCKERCONFIG = 'eyJhdXRocyI6IFtdfQ=='; // base64 for '{"auths": []}'
 
 export class DockerConfigApiService implements IDockerConfigApi {
-  private readonly coreV1API: CoreV1API;
+  private readonly coreV1API: k8s.CoreV1Api;
 
   constructor(kc: k8s.KubeConfig) {
-    this.coreV1API = prepareCoreV1API(kc);
+    this.coreV1API = kc.makeApiClient(k8s.CoreV1Api);
   }
 
-  async read(namespace: string): Promise<api.IDockerConfig> {
+  async read(namespace: string): Promise<string> {
     try {
-      const secret = await this.coreV1API.readNamespacedSecret(SECRET_NAME, namespace);
-      return this.toDockerConfig(secret.body);
+      const { body } = await this.coreV1API.readNamespacedSecret(SECRET_NAME, namespace);
+      return this.getDockerConfig(body);
     } catch (error) {
-      if (helpers.errors.isKubeClientError(error) && error.statusCode === 404) {
-        return this.toDockerConfig();
+      if ((error as HttpError)?.statusCode === 404) {
```

Comment:
Does it mean that error is not a kubeclient error or detecting kubeclienterror is incorrect?